src/main/java/com/google/sps/data/CalendarAccess.java

```diff
@@ -0,0 +1,91 @@
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.DateTime;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.CalendarScopes;
+import com.google.api.services.calendar.model.ConferenceData;
+import com.google.api.services.calendar.model.ConferenceSolution;
+import com.google.api.services.calendar.model.ConferenceSolutionKey;
+import com.google.api.services.calendar.model.CreateConferenceRequest;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.EventAttendee;
+import com.google.api.services.calendar.model.EventDateTime;
+import com.google.api.services.calendar.model.EventReminder;
+import com.google.api.services.calendar.model.Events;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.util.Collections;
+
+// Handles all things Google Calendar (for now just getting a Meet link).
+public class CalendarAccess {
+  private Calendar service;
+  private static final String CALENDAR_ID = "info@jqed.dev";
+
+  // TODO: remember to write tests in the code that calls CalendarAccess() that handle what happens
+  // per each exception
+  public CalendarAccess()
+      throws GeneralSecurityException, IOException, URISyntaxException, Exception {
+    String key = new SecretFetcher("interview-me-step-2020").getSecretValue("SERVICE_ACCT_KEY");
+    GoogleCredential credential =
+        GoogleCredential.fromStream(new ByteArrayInputStream(key.getBytes()))
+            .createScoped(Collections.singletonList(CalendarScopes.CALENDAR));
+
+    service =
+        new Calendar.Builder(
+                GoogleNetHttpTransport.newTrustedTransport(),
+                JacksonFactory.getDefaultInstance(),
+                credential)
+            .setApplicationName("Interview Me CalendarAccess")
+            .build();
+  }
+
+  // Makes an event in the calendar CALENDAR_ID and returns the Meet Link associated with that
+  // event.
+  public String getMeetLink(ScheduledInterview interview)
+      throws IOException, GeneralSecurityException {
+    Event event =
+        new Event()
+            .setSummary("Interview")
+            .setDescription(
+                "This event won't be shown to users, just used to \"reserve\" a Meet link.");
+
+    DateTime startDateTime = new DateTime(interview.when().start().toString());
+    EventDateTime start = new EventDateTime().setDateTime(startDateTime);
+    System.out.println(start.getTimeZone());
+    event.setStart(start);
+
+    DateTime endDateTime = new DateTime(interview.when().end().toString());
+    EventDateTime end = new EventDateTime().setDateTime(endDateTime);
+    event.setEnd(end);
+
+    CreateConferenceRequest createRequest = new CreateConferenceRequest();
+    createRequest.setRequestId("randomstring");
```

Comment:
https://developers.google.com/resources/api-libraries/documentation/calendar/v3/java/latest/com/google/api/services/calendar/model/CreateConferenceRequest.html#setRequestId-java.lang.String-
Really don't know if we need to regenerate this each time. ln the test environment, l called this function twice in a row and 2 events popped up, even though the same requestId was being used.

Reply:
It says
> If an ID provided is the same as for the previous request, the request is ignored.
 
If the request is not idempotent when the same id is passed, then the API is lying to you. 

(probably best not to keep it as this fake string though)